src/Core/Localization/CLDR/Currency.php

```diff
@@ -0,0 +1,183 @@
+        $this->numericIsoCode = $currencyData->getNumericIsoCode();
+        $this->decimalDigits = $currencyData->getDecimalDigits();
+        $this->displayNames = $currencyData->getDisplayNames();
+        $this->symbols = $currencyData->getSymbols();
+    }
+
+    /**
+     * Get the ISO code of this currency.
+     *
+     * @return string
+     *                The currency's ISO 4217 code
+     */
+    public function getIsoCode()
+    {
+        return $this->isoCode;
+    }
+
+    /**
+     * Get the numeric ISO code of this currency.
+     *
+     * @return string
+     *                The currency's ISO 4217 numeric code
+     */
+    public function getNumericIsoCode()
+    {
+        return $this->numericIsoCode;
+    }
+
+    /**
+     * Get the number of decimal digits to display when formatting a price with this currency.
+     *
+     * @return int
+     *             The number of decimal digits to display
+     */
+    public function getDecimalDigits()
+    {
+        return $this->decimalDigits;
+    }
+
+    /**
+     * Get the display name for the passed count context.
+     *
+     * @param string $countContext
+     *                             The count context
+     *                             "default" = talking about the currency (e.g.: "used currency is Euro")
+     *                             "one"     = talking about one unit of this currency (e.g.: "one euro")
+     *                             "other"   = talking about several units of this currency (e.g.: "ten euros")
+     *
+     * @return string
+     *                The wanted display name
+     */
+    public function getDisplayName($countContext = CurrencyInterface::DISPLAY_NAME_COUNT_DEFAULT)
+    {
+        if (!in_array($countContext, [CurrencyInterface::DISPLAY_NAME_COUNT_DEFAULT, CurrencyInterface::DISPLAY_NAME_COUNT_ONE, CurrencyInterface::DISPLAY_NAME_COUNT_OTHER])) {
+            throw new LocalizationException(
+                sprintf(
+                    'Unknown display name: "%s"',
+                    print_r($countContext, true)
+                )
+            );
+        }
+
+        return $this->displayNames[$countContext];
+    }
+
+    /**
+     * Get the symbol of this currency. Narrow symbol is returned by default.
+     *
+     * @param string $type
+     *                     Possible value: "default" ("$") and "narrow" ("US$")
+     *
+     * @return string
+     *                The currency's symbol
+     *
+     * @throws LocalizationException
+     *                               When an invalid symbol type is passed
+     */
+    public function getSymbol($type = CurrencyInterface::SYMBOL_TYPE_NARROW)
+    {
+        if (!in_array($type, [CurrencyInterface::SYMBOL_TYPE_NARROW, CurrencyInterface::SYMBOL_TYPE_DEFAULT])) {
+            throw new LocalizationException(
+                sprintf(
+                    'Unknown symbol type: "%s"',
+                    print_r($type, true)
```

Comment:
No need to print_r a string value.

Reply:
since we are not in php 7+ input is not locked as a string ;)

Reply:
well in this case prefer the annotation `(string) $countContext`

Reply:
what if given $countContext is an array ? ^^